doc/branches-explained.md

```diff
@@ -0,0 +1,39 @@
+# HDF5 Git Branching Model Explained
+
+This document desribes current HDF5 branches. 
+
+Branches are tested nightly and testing results are available at https://cdash-internal.hdfgroup.org/ and https://cdash.hdfgroup.org/. Commits that break daily testing should be fixed by 3:00 pm Central time or reverted.  We encourgae code contributors to check the status of their commits. If you have any questions, please conrtact lrknox@hdfgroup.org.
```

Comment:
Typo 'encourgae'. Should we hardcode Larry's email or have something more generic?

Reply:
Done.